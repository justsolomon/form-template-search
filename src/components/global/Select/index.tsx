import { ReactComponent as DropdownIcon } from 'assets/vectors/arrow-down.svg';
import styles from './select.module.scss';

interface SelectProps extends React.SelectHTMLAttributes<HTMLSelectElement> {
  label: string;
  options: string[];
}

const Select = ({ label, options, ...rest }: SelectProps) => {
  return (
    <div className={styles['select-group']}>
      <label className={styles['select-group__label']}>{label}</label>

      <select
        className={styles['select-group__select']}
        {...rest}
        data-testid="select"
      >
        {options.map((option, index) => {
          return (
            <option value={option} key={index}>
              {option}
            </option>
          );
        })}
      </select>

      <div className={styles['select-group__dropdown-icon']}>
        <DropdownIcon />
      </div>
    </div>
  );
};

export default Select;
