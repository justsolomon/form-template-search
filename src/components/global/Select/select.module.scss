@import 'styles/variables';
@import 'styles/mixins';

.select-group {
  @include flex-center();

  justify-content: flex-start;
  position: relative;

  &__label {
    position: absolute;
    left: 18px;
    top: -7px;
    padding: 0 6px;
    font-size: $font-size--xs;
    color: $secondary-color;
    background-color: $white;
    z-index: 1;
  }

  &__select {
    height: 40px;
    width: 100%;
    padding: 0 24px;
    border: 1px solid $select-border;
    appearance: none;
    cursor: pointer;
    border-radius: $border-radius--sm;
    background-color: $white;
  }

  &__dropdown-icon {
    @include input-group-element();

    padding: 0 15px;
    left: unset;
    right: 0;

    svg {
      height: 6px;
      width: 12px;
    }
  }

  @media #{$media-mobile--sm} {
    &__label {
      left: 10px;
    }

    &__select {
      padding: 0 16px;
    }
  }
}
