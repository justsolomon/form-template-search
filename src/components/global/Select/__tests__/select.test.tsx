import { render, screen } from '@testing-library/react';
import Select from '..';

describe('<Select />', () => {
  it('should render the Select component correctly', () => {
    const { asFragment } = render(
      <Select label="Test" options={['TestOption1', 'TestOption2']}>
        Hey
      </Select>,
    );

    expect(asFragment).toMatchSnapshot();
  });
  it('should render all the options passed into it', () => {
    render(
      <Select label="Test" options={['TestOption1', 'TestOption2']}>
        Hey
      </Select>,
    );

    const selectInput = screen.queryByTestId('select');
    expect(selectInput?.querySelectorAll('option').length).toBe(2);
  });
});
