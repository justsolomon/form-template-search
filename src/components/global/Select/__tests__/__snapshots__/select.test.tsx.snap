// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Select /> should render the Select component correctly 1`] = `[Function]`;
